### ISSUE TITLE
Suggestion: wamp.error.protocol_error

### ISSUE THREAD

**Issue by FGasper** [wamp-proto/wamp-proto#281]
It would be useful to have a default “Error” URI for protocol errors. It would also be useful if the protocol spec illustrated a standard “explanation”, e.g.:
```
[6, { "reason":"duplicate HELLO" }, "wamp.error.protocol_error"]

**Comment by volks73** [comment]
I am working on a side project for my own learning purposes where I would like to implement the WAMP protocol. I have been reviewing the current specification, and I was wondering about the handling of protocol errors. I did not see any reason URIs associated with protocol errors, other than possibly `wamp.error.no_such_realm`. I found this issue, and I too believe a protocol error or reason is needed. However, I would like to propose the `wamp.error.protocol` URI, as the `wamp.error.protocol_error` is a little redundant to me. 

I would also like to share some additional thoughts/ideas after thinking about this a little bit:

The examples given in the Protocol Errors section (5.3.3) of the specification can be reduced to a single reason: `wamp.error.unexpected_message`. A message, like either "More than one HELLO message was encountered during the current session's lifetime" or "An implicitly and clearly unexpected message was received for the current state", could be sent back to the peer using the Details component of the ABORT and GOODBYE messages depending on the state of the session.

This leads to some other ideas/developments related to a `wamp.error.protocol` and/or `wamp.error.unexpected_message` URI:

The `wamp.error.unexpected_message` URI could be used during session establishment if a message is received while the router/broker/dealer is in the process of establishing the session. In other words, the HELLO message has been sent, but the peer sends another message _before_ the WELCOME or ABORT messages have been sent. It is not clear to me if an ABORT message should be sent back or if the message should simply be ignored during this state? Sending an ABORT message seems more appropriate so that the peer knows why the message was not handled, but then what is the reason? Is this a `wamp.error.network_failure`? Well, it is actually an unexpected message and the `wamp.error.unexpected_message` URI would be appropriate.

The `wamp.error.unexpected_message` URI/reason could also be useful if a broker-only implementation receives a dealer-specific message and vice versa. However, the `wamp.error.no_such_role` reason might be more appropriate with these situations. It is a little unclear to me the appropriate response/action for a router/dealer/broker when it receives a message defined within the specification/protocol but not implemented/supported.
